src/ctapipe/utils/astro.py

```diff
@@ -4,63 +4,218 @@
 not provided by external packages and/or to satisfy particular needs of
 usage within ctapipe.
 """
-from astropy import units as u
+
+import logging
+from enum import Enum
+
+import astropy.units as u
 from astropy.coordinates import Angle, SkyCoord
+from astropy.time import Time
+
+log = logging.getLogger("main")
+
+__all__ = ["get_bright_stars", "get_hipparcos_stars"]
+
+
+class StarCatalogues(Enum):
+    Yale = {
+        "directory": "V/50/catalog",
+        "band": ["V"],
+        "ID_type": "HR",
+    }  #: Yale bright star catalogue
+    Hipparcos = {
+        "directory": "I/239/hip_main",
+        "band": ["V", "B"],
+        "ID_type": "HIP",
+    }  #: hipparcos catalogue
+
+
+def select_stars(stars, pointing=None, radius=None, magnitude_cut=None, band="B"):
+    """
+    utility function to cut stars based on magnitude and/or location
+
+    Parameters
+    ----------
+    stars: astropy table
+        Table of stars, including magnitude and coordinates
+    pointing: astropy Skycoord
+        pointing direction in the sky (if none is given, full sky is returned)
+    radius: astropy angular units
+        Radius of the sky region around pointing position. Default: full sky
+    magnitude_cut: float
+        Return only stars above a given apparent magnitude. Default: None (all entries)
+    band: string
+        wavelength band to use for the magnitude cut options are V and B. Default: 'B' (all entries)
+
+    Returns
+    -------
+    Astropy table:
+       List of all stars after cuts with same keys as the input table stars
+    """
+
+    if magnitude_cut is not None:
+        if band == "B":
+            if "Bmag" in stars.keys():
+                stars = stars[stars["Bmag"] < magnitude_cut]
+            elif "BTmag" in stars.keys():
+                stars = stars[stars["BTmag"] < magnitude_cut]
+            else:
+                raise ValueError(
+                    "The requested catalogue has no compatible magnitude for the B band"
+                )
 
-__all__ = ["get_bright_stars"]
+        if band == "V":
+            if "Vmag" in stars.keys():
+                stars = stars[stars["Vmag"] < magnitude_cut]
+            elif "VTmag" in stars.keys():
+                stars = stars[stars["VTmag"] < magnitude_cut]
+            else:
+                raise ValueError(
+                    "The requested catalogue has no compatible magnitude for the V band"
+                )
+
+    if radius is not None:
+        if pointing is None:
+            raise ValueError(
+                "Sky pointing, pointing=SkyCoord(), must be "
+                "provided if radius is given."
+            )
+        separations = stars["ra_dec"].separation(pointing)
+        stars["separation"] = separations
+        stars = stars[separations < radius]
 
+    return stars
 
-def get_bright_stars(pointing=None, radius=None, magnitude_cut=None):
+
+def get_star_catalog(catalog, min_magnitude=0.0, max_magnitude=10.0, row_limit=1000000):
```

Comment:
As commented elsewhere, I don't think we should use a min_magnitude at all. The condition can just be `f"< {max_maginitude}"`